nixos.nix

```
@@ -88,15 +88,66 @@ in
     system.activationScripts =
       let
         # Create a directory in persistent storage, so we can bind
-        # mount it.
-        mkDirCreationSnippet = persistentStoragePath: dir:
-          let
-            targetDir = concatPaths [ persistentStoragePath dir ];
-          in
+        # mount it. The directory structure's mode and ownership mirror those of
+        # persistentStoragePath/dir;
+        # TODO: Move this script to it's own file, add CI with shfmt/shellcheck.
+        mkDirWithModeAndPerms = persistentStoragePath: dir:
```

```suggestion
        mkDirWithPerms = persistentStoragePath: dir:
```

I'd argue that having both "mode" and "perms" is redundant, since the mode is (IMO) a way to set permissions ("Can others read, write, or execute this? Yes; I permit others to read and execute this."). The function name becomes less of a mouthful if we shorten it to the above suggestion.

---

Done in separate commit.